File: DatabindingWithNewsApi/app/src/main/java/com/enpassio/databindingwithnewsapi/MainActivity.java
package com.enpassio.databindingwithnewsapi;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        NewsRepository mRepo = NewsRepository.getInstance();
        mRepo.getArticles().observe(this, articles -> {
            if (articles != null) {
                for (Article article : articles) {
                    Log.d(TAG, "article: " + article.toString());
                }
            }
        });
    }
}

File: DatabindingWithNewsApiKotlin/app/src/main/java/com/enpassio/databindingwithnewsapikotlin/ui/MainActivity.kt
package com.enpassio.databindingwithnewsapikotlin.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.transaction
import com.enpassio.databindingwithnewsapikotlin.R


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            supportFragmentManager.transaction {
                add(R.id.fragment_holder, ArticleListFragment())
            }
        }
    }
}
